### ISSUE TITLE
Fix files on external storages

### ISSUE THREAD

**Issue by AlessioP3** [nextcloud-gmbh/mtime_fixer_tool_kit#17]
Hi,
I have run the unsolvable_files.sh script.
The script lists me a list of files, but at this point I don't know how to proceed to fix the files

All these files are on an external drive connected via USB.

Thank you in advance.

**Comment by artonge** [comment]
How is Nextcloud connected to them? External storage?

**Comment by AlessioP3** [comment]
Yes, I connected the external drive to nextcloud via the nextcloud application called "external storage support"

**Comment by artonge** [comment]
The script does not support that currently. Feel free to make a pull request to add support for external storage :)

**Comment by AlessioP3** [comment]
Thank you for your reply.
Could you please explain what the problem is with these files and tell me theoretically what the script should do to correct the files?

**Comment by artonge** [comment]
The problem is that the script is not currently able to know where they are located.

While in theory this could be done, it is probably easier for you to take the output of the script and run `touch` on every listed file.

**Comment by AlessioP3** [comment]
thank you for the suggestion, I ran touch on the files and the problem was solved.
After a few days, however, the problem returned on other files.

I have identified the client that is generating the problem every time it modifies a file.
In order to understand how to solve the root of the problem, can you tell me what is causing the error?

**Comment by artonge** [comment]
The desktop client version 3.4.0 included several fixes for the Virtual files feature on Windows. Unfortunately one of those fixes had a side effect that removes the modification time from files and trigger an upload to the server of those files with invalid metadata.

The latest updates of both the client and the server should prevent this behaviour.

**Comment by jahill92** [comment]
Hello @artonge 
Any idea on how to address this issue for external Samba stores?
I have a central external smb store that every user has access to and a user dir external storage configured with $user as share.
Since the SMB access is not transparent to the system I cant run touch on the files.
